crud/common/utils.lua

```diff
@@ -711,4 +711,41 @@ function utils.update_storage_call_error_description(err, func_name, replicaset_
     return err
 end
 
+--- Insert each value from values to list
+--
+-- @function list_extend
+--
+-- @param table list
+--  List to be extended
+--
+-- @param table values
+--  Values to be inserted to list
+--
+-- @return[1] list
+--  List with old values and inserted values
+function utils.list_extend(list, values)
+    dev_checks('table', 'table')
+
+    for _, value in ipairs(values) do
+        table.insert(list, value)
+    end
+
+    return list
+end
+
+function utils.list_slice(list, start_index, end_index)
+    dev_checks('table', 'number', '?number')
+
+    if end_index == nil then
+        end_index = utils.table_count(list)
+    end
```

Comment:
Nit: if the list is assumed to be a list (a table with numeric keys starting from 1 and without holes), we can just use `#list`. If it is the list with holes (with numeric keys starting from 1), we can use `table.maxn(list)`.

```lua
tarantool> clock = require('clock')
tarantool> t = {}
tarantool> for i = 1, 1000 do if i % 2 == 1 then t[i] = i end end
tarantool> function table_count(table)
         >     local cnt = 0
         >     for _, _ in pairs(table) do
         >         cnt = cnt + 1
         >     end
         > 
         >     return cnt
         > end
tarantool> clock.bench(function() for i = 1, 10^6 do table.maxn(t) end end)
---
- - 0.033955832
...
tarantool> clock.bench(function() for i = 1, 10^6 do table_count(t) end end)
---
- - 1.834358107
...
```

If we know that the table is the array without holes:

```lua

tarantool> t = {}
tarantool> for i = 1, 1000 do t[i] = i end
tarantool> clock.bench(function() for i = 1, 10^6 do table.maxn(t) end end)
---
- - 0.041307265
...
tarantool> clock.bench(function() for i = 1, 10^6 do table_count(t) end end)
---
- - 3.240869489
...
tarantool> clock.bench(function() for i = 1, 10^6 do local x = #t end end)
---
- - 0.00034222800000006
...
```

Reply:
Updated.